include/Jit/LLILCJit.h

```diff
@@ -167,6 +186,10 @@ struct LLILCJitPerThreadState {
 /// top-level invocations of the jit is held in thread local storage.
 class LLILCJit : public ICorJitCompiler {
 public:
+  typedef llvm::orc::ObjectLinkingLayer<> LoadLayerT;
+  typedef llvm::orc::IRCompileLayer<LoadLayerT> CompileLayerT;
+  typedef CompileLayerT::ModuleSetHandleT ModuleHandleT;
```

Comment:
While I'm not a fan of terminal `T`s in general, I really think it's helpful to have them here, and yes that's because the ORC code uses them.  The ORC code makes heavy use of type parameterization, using the terminal `T` to signal both the type parameters to templates and the exported typedefs like these that make the type expressions more manageable (both by reducing their size and by abstracting away implementation details of lower layers, which are typically template arguments of higher layers).  Also I think in the case of typedefs for layers, there's a good chance that the name without the T would be the same as the template name; following convention here avoids any such conflicts (which didn't happen here, but may happen as we add layers, and maybe I should have used `ObjectLinkingLayerT` and `IRCompileLayerT`...).
Taking a closer look at this, though, made me realize that the ModuleHandleT I defined was vestigial and nowhere referenced, so I'll push up an update that omits it.